rest-api/src/main/java/com/cloudbees/workflow/rest/external/RunExt.java

```diff
@@ -332,8 +332,8 @@ public static RunExt createOld(WorkflowRun run) {
             if (artifacts != null && !artifacts.isEmpty()) {
                 runExt.get_links().setArtifacts(Link.newLink(RunAPI.getArtifactsUrl(run)));
             }
-
             FlowGraphWalker walker = new FlowGraphWalker(execution);
+            List<FlowNode> sortedNodes = FlowNodeUtil.getIdSortedExecutionNodeList(execution); // Hold a ref to prevent GC until we're done analyzing
```

Comment:
Right. These days they are pretty smart. And I've seen things optimized out that were even referenced in parts of unused branch expressions. Anyway, as you say, it seems the fallback(s) will at least let this continue to function in this case. I was just noting it to be reviewed subsequently.